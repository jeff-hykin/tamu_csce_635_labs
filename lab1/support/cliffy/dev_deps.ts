/* std */
export {
  assert,
  assertEquals,
  assertInstanceOf,
  assertRejects,
  assertStrictEquals,
  assertThrows,
} from "https://deno.land/std@0.196.0/testing/asserts.ts";
export {
  assertSpyCall,
  assertSpyCalls,
  spy,
} from "https://deno.land/std@0.196.0/testing/mock.ts";
export { assertSnapshot } from "https://deno.land/std@0.196.0/testing/snapshot.ts";
export { describe, it } from "https://deno.land/std@0.196.0/testing/bdd.ts";
export {
  assertType,
  type IsExact,
} from "https://deno.land/std@0.196.0/testing/types.ts";
export {
  bold,
  red,
  stripColor,
} from "https://deno.land/std@0.196.0/fmt/colors.ts";
export { dirname } from "https://deno.land/std@0.196.0/path/mod.ts";
export { expandGlob } from "https://deno.land/std@0.196.0/fs/expand_glob.ts";
export type { WalkEntry } from "https://deno.land/std@0.196.0/fs/walk.ts";
export { copy } from "https://deno.land/std@0.196.0/streams/copy.ts";
export { format } from "https://deno.land/std@0.196.0/datetime/format.ts";
export { gt } from "https://deno.land/std@0.196.0/semver/gt.ts";
export { lt } from "https://deno.land/std@0.196.0/semver/lt.ts";

/* 3rd party */
export { default as sinon } from "https://cdn.skypack.dev/sinon@v13.0.2?dts";
