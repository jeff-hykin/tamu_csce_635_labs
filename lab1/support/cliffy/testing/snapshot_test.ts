import { assert, assertSnapshot } from "../dev_deps.ts";
import { quoteString } from "./_quote_string.ts";
import { dirname, fromFileUrl } from "./deps.ts";

Deno.test({
  name: "should run snapshot tests",
  async fn(ctx) {
    const testDir = dirname(fromFileUrl(import.meta.url));
    const snapshotDir = testDir + "/__snapshots__";
    const snapshotTestDir = testDir + "/__snapshots_test__";

    const snapshotPath = `${snapshotDir}/snapshot_test_fixture.ts.snap`;
    const snapshot2Path = `${snapshotDir}/snapshot_test_fixture_2.ts.snap`;
    const snapshot3Path = `${snapshotTestDir}/snapshot_test_fixture.ts.snap`;

    const args = [
      "test",
      "--allow-run=deno",
      `--allow-read=${testDir}`,
      `--allow-write=${testDir}`,
      "testing/snapshot_test_fixture.ts",
      "--",
      "--update",
    ];

    const cmd = new Deno.Command("deno", { args });

    const { success, stdout, stderr } = await cmd.output();

    const decoder = new TextDecoder();
    assert(success, decoder.decode(stderr) + decoder.decode(stdout));

    const snapshotContent = await Deno.readTextFile(snapshotPath);
    const snapshot2Content = await Deno.readTextFile(snapshot2Path);
    const snapshot3Content = await Deno.readTextFile(snapshot3Path);

    await assertSnapshot(ctx, snapshotContent, { serializer: quoteString });
    await assertSnapshot(ctx, snapshot2Content, { serializer: quoteString });
    await assertSnapshot(ctx, snapshot3Content, { serializer: quoteString });

    await Deno.remove(snapshotPath);
    await Deno.remove(snapshot2Path);
    await Deno.remove(snapshotTestDir, { recursive: true });
  },
});
