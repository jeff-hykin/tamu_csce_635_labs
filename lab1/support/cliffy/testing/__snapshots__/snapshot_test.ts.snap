export const snapshot = {};

snapshot[`should run snapshot tests 1`] = `
"export const snapshot = {};

snapshot[\`should create a simple snapshot 1\`] = \`
stdout:
\\"foo
\\"
stderr:
\\"bar
\\"
\`;

snapshot[\`should set deno args > step 1 1\`] = \`
stdout:
'[ \\"foo\\", \\"bar\\", \\"beep\\" ]
'
stderr:
\\"\\"
\`;

snapshot[\`should set deno args > step 2 1\`] = \`
stdout:
'[ \\"foo\\", \\"bar\\", \\"boop\\" ]
'
stderr:
\\"\\"
\`;

snapshot[\`should write to stdin > step 1 1\`] = \`
stdout:
\\"# stdin start
foo
# stdin end
\\"
stderr:
\\"\\"
\`;

snapshot[\`should write to stdin > step 2 1\`] = \`
stdout:
\\"# stdin start
bar
# stdin end
\\"
stderr:
\\"\\"
\`;
"
`;

snapshot[`should run snapshot tests 2`] = `
'export const snapshot = {};

snapshot[\`should change the snapshot path 1\`] = \`
stdout:
"foo
"
stderr:
"bar
"
\`;
'
`;

snapshot[`should run snapshot tests 3`] = `
'export const snapshot = {};

snapshot[\`should change the snapshot dir 1\`] = \`
stdout:
"foo
"
stderr:
"bar
"
\`;
'
`;
